src/jdk.jpackage/share/classes/jdk/jpackage/internal/AppImageFile.java

```diff
@@ -61,10 +63,11 @@ public final class AppImageFile {
     // These values will be loaded from AppImage xml file.
     private final String creatorVersion;
     private final String creatorPlatform;
+    private final String appVersion;
     private final String launcherName;
     private final String mainClass;
     private final List<LauncherInfo> addLauncherInfos;
-    private final boolean signed;
```

Comment:
This change defeats the idea of keeping `AppImageFile` immutable. I'd rather add `AppImageFile.copyAsSigned()` method that will return the same `AppImageFile` instance if `signed` is `true`, or a copy with `signed` set to `true` otherwise.

I'd add `public AppImageFile.save(Path appImageDir)` member function to serialize `AppImageFile` instance and keep the signature of existing static `AppImageFile.save` unchanged. I'd make them call static `private AppImageFile.save(Path appImageDir, Map<String, Object> params, AppImageFile appImage)` that you have right now in the review. `AppImageFile.save(Path appImageDir, Map<String, Object> params, AppImageFile appImage)` is implementation details and should not be a part of class public interface.

Reply:
8293462: [macos] app image signature invalid when creating DMG or PKG from post processed signed image [v2]
 - Fixed as suggested.
 - Moved getAttribute() under LauncherInfo class, since it is used only by this class.
 - Added unit test for AppImageFile.copyAsSigned().